habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py

```diff
@@ -309,13 +347,204 @@ def load_objects(sim, objects):
+) -> List[ObjectViewLocation]:
+    assert obj is not None
+
+    cached_obj_transform = obj.transformation
+    if object_transform:
+        obj.transformation = object_transform
+
+    object_id = obj.object_id
+    object_aabb = get_aabb(object_id, sim, transformed=True)
+    object_position = object_aabb.center()
+
+    center = np.array(obj.translation)
+    sizes = np.array(obj.root_scene_node.cumulative_bb.size())
+    rotation = obj.rotation
+    object_obb = habitat_sim.geo.OBB(center, sizes, rotation)
+
+    eps = 1e-5
+
+    object_nodes = obj.visual_scene_nodes
+    assert all(node.semantic_id == object_id + 1 for node in object_nodes)
+    semantic_id = object_nodes[0].semantic_id
+
+    object_nodes = obj.visual_scene_nodes
+    assert all(node.semantic_id == object_id + 1 for node in object_nodes)
+    semantic_id = object_nodes[0].semantic_id
+
+    max_distance = 1.0
+    cell_size = 0.3 / 2.0
+    x_len, _, z_len = object_aabb.size() / 2.0 + mn.Vector3(max_distance)
+    x_bxp = np.arange(-x_len, x_len + eps, step=cell_size) + object_position[0]
+    z_bxp = np.arange(-z_len, z_len + eps, step=cell_size) + object_position[2]
+    candidate_poses = [
+        np.array([x, object_position[1], z])
+        for x, z in itertools.product(x_bxp, z_bxp)
+    ]
+
+    def down_is_navigable(pt, search_dist=2.0):
+        pf = sim.pathfinder
+        delta_y = 0.05
+        max_steps = int(search_dist / delta_y)
+        step = 0
+        is_navigable = pf.is_navigable(pt, 2)
+        while not is_navigable:
+            pt[1] -= delta_y
+            is_navigable = pf.is_navigable(pt)
+            step += 1
+            if step == max_steps:
+                return False
+        return True
+
+    def _get_iou(x, y, z):
+        pt = np.array([x, y, z])
+
+        if not object_obb.distance(pt) <= max_distance:
+            return -0.5, pt, None, "Unknown error"
+
+        if not down_is_navigable(pt):
+            return -1.0, pt, None, "Down is not navigable"
+        pf = sim.pathfinder
+        pt = np.array(pf.snap_point(pt))
+        pt[1] += pf.nav_mesh_settings.agent_height
+
+        goal_direction = object_position - pt
+        goal_direction[1] = 0
+
+        q = direction_to_quaternion(goal_direction)
+
+        cov = 0
+        agent = sim.get_agent(0)
+        for act in [
+            "look_down",
+            "look_up",
+            "look_up",
+        ]:
+            agent.act(act)
+            for v in agent._sensors.values():
+                v.set_transformation_from_spec()
+            agent_state = agent.get_state()
+            agent_state.position = pt
+            agent_state.rotation = q
+            agent.set_state(agent_state, reset_sensors=False)
+            obs = sim.get_sensor_observations(0)
+            cov += compute_pixel_coverage(obs["semantic"], semantic_id)
+
+        # from habitat.utils.visualizations.utils import observations_to_image
```

Comment:
Maybe add a `debug` flag to support this visualization instead of removing this

Reply:
Added this and adapted it from original object nav episode generation to make it work now.